From ac1754f173ab774c63102470f576b688d0f407ad Mon Sep 17 00:00:00 2001
From: Rakesh Kota <quic_kotarake@quicinc.com>
Date: Wed, 23 Oct 2024 22:29:46 +0530
Subject: [PATCH] PENDING: arm64: dts: qcom: Add support for PMIC peripherals
 for qcs8300

Add support for MBG TM, PON and ADC peripherals for qcs8300.

Signed-off-by: Rakesh Kota <quic_kotarake@quicinc.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/qcom/qcs8300-pmics.dtsi | 352 ++++++++++++++++++++
 1 file changed, 352 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs8300-pmics.dtsi b/arch/arm64/boot/dts/qcom/qcs8300-pmics.dtsi
index a94b0bfa98dc..2c4d93bf703e 100644
--- a/arch/arm64/boot/dts/qcom/qcs8300-pmics.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs8300-pmics.dtsi
@@ -3,9 +3,170 @@
  * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
+#include <dt-bindings/iio/qcom,spmi-adc5-gen3-pm8775.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/spmi/spmi.h>
 
+/ {
+	thermal-zones {
+		pmm8620au_0_mbg_tz {
+			polling-delay-passive = <100>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_sail_0_tz>;
+
+			trips {
+				trip0 {
+					temperature = <115000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <135000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+			};
+		};
+
+		pmm8620au-0-therm {
+			polling-delay-passive = <100>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_0_temp_alarm>;
+
+			trips {
+				trip0 {
+					temperature = <105000>;
+					hysteresis = <0>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <0>;
+					type = "critical";
+				};
+			};
+		};
+
+		pmm8620au_0_sail_xo {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_0_adc PM8775_1_ADC5_GEN3_AMUX6_THM_100K_PU>;
+			trips {
+				trip0 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
+
+		pmm8620au_0_ufs0 {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_0_adc PM8775_1_ADC5_GEN3_AMUX5_THM_100K_PU>;
+			trips {
+				trip0 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
+
+		pmm8620au_0_xo {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_0_adc PM8775_1_ADC5_GEN3_AMUX1_THM_100K_PU>;
+			trips {
+				trip0 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
+
+		pmm8650au_1_mbg_tz {
+			polling-delay-passive = <100>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8620au_sail_1_tz>;
+
+			trips {
+				trip0 {
+					temperature = <115000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <135000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+			};
+		};
+
+		pmm8650au-1-therm {
+			polling-delay-passive = <100>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8650au_1_temp_alarm>;
+
+			trips {
+				trip0 {
+					temperature = <105000>;
+					hysteresis = <0>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <0>;
+					type = "critical";
+				};
+			};
+		};
+
+		pmm8650au_1_sdram {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmm8650au_1_adc PM8775_2_ADC5_GEN3_AMUX6_THM_100K_PU>;
+			trips {
+				trip0 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+
+				trip1 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
+	};
+};
+
 &spmi_bus {
 	pmm8620au_0: pmic@0 {
 		compatible = "qcom,pmm8654au", "qcom,spmi-pmic";
@@ -13,6 +174,98 @@ pmm8620au_0: pmic@0 {
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		pmm8620au_0_adc: vadc@8000 {
+			compatible = "qcom,spmi-adc5-gen3";
+			reg = <0x8000>;
+			reg-names = "adc5-gen3-base";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts-extended = <&spmi_bus 0x0 0x80 0x1 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "adc-sdam0";
+			#io-channel-cells = <1>;
+			#thermal-sensor-cells = <1>;
+
+			pmm8620au_0_ref_gnd {
+				reg = <PM8775_1_ADC5_GEN3_OFFSET_REF>;
+				label = "pmm8620au_0_ref_gnd";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8620au_0_vref_1p25 {
+				reg = <PM8775_1_ADC5_GEN3_1P25VREF>;
+				label = "pmm8620au_0_vref_1p25";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8620au_0_die_temp {
+				reg = <PM8775_1_ADC5_GEN3_DIE_TEMP>;
+				label = "pmm8620au_0_die_temp";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8620au_0_vph_pwr {
+				reg = <PM8775_1_ADC5_GEN3_VPH_PWR>;
+				label = "pmm8620au_0_vph_pwr";
+				qcom,pre-scaling = <1 3>;
+			};
+
+			pmm8620au_0_sail_xo_therm {
+				reg = <PM8775_1_ADC5_GEN3_AMUX6_THM_100K_PU>;
+				label = "pmm8620au_0_sail_xo_therm";
+				qcom,ratiometric;
+				qcom,hw-settle-time = <200>;
+				qcom,adc-tm;
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8620au_0_ufs0_therm {
+				reg = <PM8775_1_ADC5_GEN3_AMUX5_THM_100K_PU>;
+				label = "pmm8620au_0_ufs0_therm";
+				qcom,ratiometric;
+				qcom,hw-settle-time = <200>;
+				qcom,adc-tm;
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8620au_0_xo_therm {
+				reg = <PM8775_1_ADC5_GEN3_AMUX1_THM_100K_PU>;
+				label = "pmm8620au_0_xo_therm";
+				qcom,ratiometric;
+				qcom,hw-settle-time = <700>;
+				qcom,adc-tm;
+				qcom,pre-scaling = <1 1>;
+			};
+		};
+
+		pmm8620au_0_temp_alarm: temp-alarm@a00 {
+			compatible = "qcom,spmi-temp-alarm";
+			reg = <0xa00>;
+			interrupts-extended = <&spmi_bus 0x0 0xa 0x0 IRQ_TYPE_EDGE_BOTH>;
+			io-channels = <&pmm8620au_0_adc PM8775_1_ADC5_GEN3_DIE_TEMP>;
+			io-channel-names = "thermal";
+			#thermal-sensor-cells = <0>;
+		};
+
+		pmm8620au_0_pon: pon@1200 {
+			compatible = "qcom,pmk8350-pon";
+			reg = <0x1200>, <0x800>;
+			reg-names = "hlos", "pbs";
+
+			pmm8620au_0_pon_pwrkey: pwrkey {
+				compatible = "qcom,pmk8350-pwrkey";
+				interrupts-extended = <&spmi_bus 0x0 0x12 0x7 IRQ_TYPE_EDGE_BOTH>;
+				linux,code = <KEY_POWER>;
+				debounce = <15625>;
+			};
+
+			pmm8620au_0_pon_resin: resin {
+				compatible = "qcom,pmk8350-resin";
+				interrupts-extended = <&spmi_bus 0x0 0x12 0x6 IRQ_TYPE_EDGE_BOTH>;
+				debounce = <15625>;
+				linux,code = <KEY_VOLUMEDOWN>;
+			};
+		};
+
 		pmm8620au_0_rtc: rtc@6100 {
 			compatible = "qcom,pmk8350-rtc";
 			reg = <0x6100>, <0x6200>;
@@ -21,6 +274,19 @@ pmm8620au_0_rtc: rtc@6100 {
 			allow-set-time;
 		};
 
+		pmm8620au_0_sdam_0: nvram@7100 {
+			compatible = "qcom,spmi-sdam";
+			reg = <0x7100>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0 0x7100 0x100>;
+
+			reboot_reason: reboot-reason@48 {
+				reg = <0x48 0x1>;
+				bits = <1 7>;
+			};
+		};
+
 		pmm8620au_0_gpios: gpio@8800 {
 			compatible = "qcom,pmm8654au-gpio", "qcom,spmi-gpio";
 			reg = <0x8800>;
@@ -32,12 +298,82 @@ pmm8620au_0_gpios: gpio@8800 {
 		};
 	};
 
+	pmm8620au_sail_0: pmic@1 {
+		compatible = "qcom,spmi-pmic";
+		reg = <1 SPMI_USID>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pmm8620au_sail_0_tz: qcom,mbg-tm@d700 {
+			compatible = "qcom,spmi-mgb-tm";
+			reg = <0xd700>;
+			io-channels = <&pmm8620au_0_adc PM8775_1_ADC5_GEN3_DIE_TEMP>;
+			io-channel-names = "thermal";
+			interrupts-extended = <&spmi_bus 0x1 0xd7 0x0 IRQ_TYPE_EDGE_RISING>;
+			#thermal-sensor-cells = <0>;
+		};
+	};
+
 	pmm8650au_1: pmic@2 {
 		compatible = "qcom,pmm8654au", "qcom,spmi-pmic";
 		reg = <0x2 SPMI_USID>;
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		pmm8650au_1_adc: vadc@8000 {
+			compatible = "qcom,spmi-adc5-gen3";
+			reg = <0x8000>;
+			reg-names = "adc5-gen3-base";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts-extended = <&spmi_bus 0x2 0x80 0x1 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "adc-sdam0";
+			#io-channel-cells = <1>;
+			#thermal-sensor-cells = <1>;
+
+			pmm8650au_1_ref_gnd {
+				reg = <PM8775_2_ADC5_GEN3_OFFSET_REF>;
+				label = "pmm8650au_1_ref_gnd";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8650au_1_vref_1p25 {
+				reg = <PM8775_2_ADC5_GEN3_1P25VREF>;
+				label = "pmm8650au_1_vref_1p25";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8650au_1_die_temp {
+				reg = <PM8775_2_ADC5_GEN3_DIE_TEMP>;
+				label = "pmm8650au_1_die_temp";
+				qcom,pre-scaling = <1 1>;
+			};
+
+			pmm8650au_1_vph_pwr {
+				reg = <PM8775_2_ADC5_GEN3_VPH_PWR>;
+				label = "pmm8650au_1_vph_pwr";
+				qcom,pre-scaling = <1 3>;
+			};
+
+			pmm8650au_1_sdram_therm {
+				reg = <PM8775_2_ADC5_GEN3_AMUX6_THM_100K_PU>;
+				label = "pmm8650au_1_sdram_therm";
+				qcom,ratiometric;
+				qcom,hw-settle-time = <200>;
+				qcom,adc-tm;
+				qcom,pre-scaling = <1 1>;
+			};
+		};
+
+		pmm8650au_1_temp_alarm: temp-alarm@a00 {
+			compatible = "qcom,spmi-temp-alarm";
+			reg = <0xa00>;
+			interrupts-extended = <&spmi_bus 0x2 0xa 0x0 IRQ_TYPE_EDGE_BOTH>;
+			io-channels = <&pmm8650au_1_adc PM8775_2_ADC5_GEN3_DIE_TEMP>;
+			io-channel-names = "thermal";
+			#thermal-sensor-cells = <0>;
+		};
+
 		pmm8650au_1_gpios: gpio@8800 {
 			compatible = "qcom,pmm8654au-gpio", "qcom,spmi-gpio";
 			reg = <0x8800>;
@@ -48,4 +384,20 @@ pmm8650au_1_gpios: gpio@8800 {
 			#interrupt-cells = <2>;
 		};
 	};
+
+	pmm8650au_sail_1: pmic@3 {
+		compatible = "qcom,spmi-pmic";
+		reg = <3 SPMI_USID>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pmm8620au_sail_1_tz: qcom,mbg-tm@d700 {
+			compatible = "qcom,spmi-mgb-tm";
+			reg = <0xd700>;
+			io-channels = <&pmm8650au_1_adc PM8775_2_ADC5_GEN3_DIE_TEMP>;
+			io-channel-names = "thermal";
+			interrupts-extended = <&spmi_bus 0x3 0xd7 0x0 IRQ_TYPE_EDGE_RISING>;
+			#thermal-sensor-cells = <0>;
+		};
+	};
 };
-- 
2.34.1

